### DISCUSSION TITLE
Assembler directive for temporarily renaming memory addresses

### DISCUSSION THREAD

**Discussion by SusiKette** [cc65/cc65#1627] [Ideas]
Many NES games seem to use a handful of bytes in the beginning of zero page as local variables and arguments for some subroutine. It would be nice to be able to refer to these memory addresses with a name that describes the purpose of that address for the current context to make writing and reading the code easier. You would give the directive the address and the name you want to use for it. Something like this:

`
 .alias $0000 controller1_read_data
`

You could still of course refer to the original name if you want to. Aliasing the same address again would overwrite the old alias.

Another use for this would be if you want to have multiple game modes and you need to use the same memory areas differently in each mode, because there is not enough for each mode to have dedicated memory space (Getsu Fuuma Den does this). Alternatively for this second use you could make multiple list of reserved memory and use another directive to switch between the two depending on which one is relevant for the code.

**Comment by mrdudz** [comment]
what about using
```
controller1_read_data = 0
```
?

**Reply by SusiKette** [reply]
Probably. I always thought that as defining constants, not as renaming variables (although I guess a constant would be the same thing)

**Comment by groessler** [comment]
Or using ".define"?

**Comment by greg-king5** [comment]
There also is the `.set` directive which allows you to reuse your "alias".
